examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,290 @@
+fig = plt.figure(figsize=(10, 10))
+gs = GridSpec(4, 2)
 
-    ax2.set_xlabel("Mean predicted value")
-    ax2.set_ylabel("Count")
-    ax2.legend(loc="upper center", ncol=2)
+ax1 = fig.add_subplot(gs[:2, :2])
+viz_objects = {}
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    viz = plot_calibration_curve(
+        clf, X_test, y_test, n_bins=10, name=name, ax=ax1, color=colors(i)
+    )
+    viz_objects[name] = viz
 
-    plt.tight_layout()
+ax1.grid()
+ax1.set_title('Calibration plots (SVC)')
 
-# Plot calibration curve for Gaussian Naive Bayes
-plot_calibration_curve(GaussianNB(), "Naive Bayes", 1)
+# Add histogram
+grid_positions = [(2, 0), (2, 1), (3, 0), (3, 1)]
+for i, (_, name) in enumerate(clf_list):
+    row, col = grid_positions[i]
+    ax = fig.add_subplot(gs[row, col])
 
-# Plot calibration curve for Linear SVC
-plot_calibration_curve(LinearSVC(max_iter=10000), "SVC", 2)
+    ax.hist(
+        viz_objects[name].y_prob, range=(0, 1), bins=10, label=name,
+        color=colors(i)
+    )
+    ax.set(title=name, xlabel="Mean predicted probability", ylabel="Count")
 
+plt.tight_layout()
 plt.show()
+
+# %%
+# :class:`~sklearn.svm.LinearSVC` shows the opposite
+# behavior to :class:`~sklearn.naive_bayes.GaussianNB`; the calibration
+# curve has a sigmoid shape, which is typical for an under-confident
+# classifier. In the case of :class:`~sklearn.svm.LinearSVC`, this is caused
+# by the margin property of the hinge loss, which lets the model focus on
+# difficult samples that are close to the decision boundary (the support
+# vectors).
+#
+# Both kinds of calibration (sigmoid and isotonic) can fix this issue and
+# yield nearly identical results.
+#
+# As before, we show the Brier score, precision, recall and F1 score below.
+
+index = []
+brier = []
+logloss = []
+precision = []
+recall = []
+f1 = []
+
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    y_proba = clf.predict_proba(X_test)
+    y_pred = clf.predict(X_test)
+
+    # Create DataFrame index
+    index.append(name)
+    # Store column data
+    brier.append(brier_score_loss(y_test, y_proba[:, 1]))
+    logloss.append(log_loss(y_test, y_proba[:, 1]))
+    precision.append(precision_score(y_test, y_pred))
+    recall.append(recall_score(y_test, y_pred))
+    f1.append(f1_score(y_test, y_pred))
+
+score_df = pd.DataFrame(
+    data={'Brier score': brier, 'Log loss': logloss, 'Precision': precision,
+          'Recall': recall, 'F1': f1},
+    index=index,
+)
+score_df.round(3)
+
+# %%
+# Summary
+# -------
+#
+# Parametric sigmoid calibration can deal with situations where the calibration
+# curve of the base classifier is sigmoid (e.g., for
+# :class:`~sklearn.svm.LinearSVC`) but not where it is transposed-sigmoid
+# (e.g., :class:`~sklearn.naive_bayes.GaussianNB`). Non-parametric
+# isotonic calibration can deal with both situations but may require more
+# data to produce good results.
```

Comment:
Empirical. From the [Niculescu-Mizil & Caruana paper](https://www.cs.cornell.edu/~alexn/papers/calibration.icml05.crc.rev3.pdf), in section 5, they look at the effect of varying the amount of calibration data on their 8 datasets and 10 models.
They also note that 'isotonic regression is less constrained than Platt scaling so it is easier for it to overfit when calibration set is small'.
I can be more careful or expand to say that it is because isotonic may overfit on small datasets?

Reply:
Still can't think of a reason. Intuitively it might be related to how the decision function values are unbounded but I couldn't find any literature on this.